engine/runtime/src/main/scala/org/enso/interpreter/runtime/SerializationManager.scala

```diff
@@ -43,32 +34,7 @@ final class SerializationManager(private val context: TruffleCompilerContext) {
   /** The debug logging level. */
   private val debugLogLevel = Level.FINE
 
-  /** A set of the modules that are currently being serialized.
-    *
-    * This set is accessed concurrently. This is safe as it is backed by a
-    * [[ConcurrentHashMap]] and is wrapped with the scala [[mutable.Set]]
-    * interface.
-    */
-  private val isSerializing: mutable.Set[QualifiedName] = buildConcurrentHashSet
-
-  /** A map of the modules awaiting serialization to their associated tasks
-    *
-    * This map is accessed concurrently.
-    */
-  private val isWaitingForSerialization =
-    collection.concurrent.TrieMap[QualifiedName, Future[Boolean]]()
-
-  /** The thread pool that handles serialization. */
-  private val pool: ThreadPoolExecutor = new ThreadPoolExecutor(
```

Comment:
This PR moves all the _threading mangling_ into `SerializationPool` to separate the work with _threads_ from the remaining logic inside of `SerializationManager`.